src/Orleans.Transactions/State/ReaderWriterLock.cs

```diff
@@ -99,6 +99,9 @@ private class LockGroup : Dictionary<Guid, TransactionRecord<TState>>
                 if (group == currentGroup)
                 {
                     group.Deadline = DateTime.UtcNow + this.options.LockTimeout;
+
+                    if (logger.IsEnabled(LogLevel.Trace))
+                        logger.Trace($"set lock expiration at {group.Deadline:o}");
```

Comment:
For all new logging we're using "semantic logging".  We've not updated exiting logging, but that's planned.

https://softwareengineering.stackexchange.com/questions/312197/benefits-of-structured-logging-vs-basic-logging

For semantic logging this would look something like.
logger.Trace("set lock expiration at {Deadline}", group.Deadline.Tostring("o"));